server/account/collection.ts

```diff
@@ -22,67 +19,135 @@ class AccountCollection {
+   *
+   * @param {string} username - The username of the credential to find
+   * @param {string} password - The password of the credential to find
+   * @return {Promise<HydratedDocument<Credential>> | Promise<null>} - The credential with the given username and password, if any
+   */
+  static async findOneByUsernameAndPassword(username: string, password: string): Promise<HydratedDocument<Credential>> {
+    return CredentialModel.findOne({
       username: new RegExp(`^${username.trim()}$`, 'i'),
       password
     });
   }
 
   /**
-   * Update an account's information
+   * Update a credential's information
    *
-   * @param {string} accountId - The id of the account to update
-   * @param {Object} accountDetails - An object with the account's updated credentials
-   * @return {Promise<HydratedDocument<Account>>} - The updated account
+   * @param {string} credentialId - The id of the credential to update
+   * @param {Object} credentialDetails - An object with the credential's updated credentials
+   * @return {Promise<HydratedDocument<Credential>>} - The updated credential
    */
-  static async updateOne(accountId: Types.ObjectId | string, accountDetails: {password?: string; username?: string}): Promise<HydratedDocument<Account>> {
-    const account = await AccountModel.findOne({_id: accountId});
-    if (accountDetails.password) {
-      account.password = accountDetails.password;
-    }
+  static async updateOne(credentialId: Types.ObjectId | string, credentialDetails: {password?: string; username?: string}): Promise<HydratedDocument<Credential>> {
+    const credential = await CredentialModel.findOne({_id: credentialId});
 
-    if (accountDetails.username) {
-      account.username = accountDetails.username;
+    if (credentialDetails.username) {
+      credential.username = credentialDetails.username;
+    }
+    if (credentialDetails.password) {
+      credential.password = credentialDetails.password;
     }
 
-    await account.save();
-    return account;
+    await credential.save();
+    return credential;
   }
 
   /**
-   * Delete an account from the collection.
+   * Delete a credential from the collection.
    *
-   * @param {string} accountId - The id of account to delete
-   * @return {Promise<boolean>} - true if the account has been deleted, false otherwise
+   * @param {string} credentialId - The id of credential to delete
    */
-  static async deleteOne(accountId: Types.ObjectId | string): Promise<boolean> {
-    const account = await AccountModel.deleteOne({_id: accountId});
-    return account !== null;
+  static async deleteOne(credentialId: Types.ObjectId | string): Promise<void> {
+    await CredentialModel.deleteOne({_id: credentialId});
   }
-}
 
-export default AccountCollection;
+  /**
+   * Delete all credentials associated with an account.
+   *
+   * @param {string} account - The id of account whose credentials to delete
+   */
+  static async deleteMany(account: Types.ObjectId): Promise<void> {
```

Comment:
just to stay consistent with the rest of the code, it might make sense to write 'accountId' in the param and update the function variables accordingly